server/src/main/java/org/apache/druid/client/cache/MemcachedCache.java

```diff
@@ -355,9 +360,23 @@ public void updateHistogram(String name, int amount)
           .setReadBufferSize(config.getReadBufferSize())
           .setOpQueueFactory(opQueueFactory)
           .setMetricCollector(metricCollector)
-          .setEnableMetrics(MetricType.DEBUG) // Not as scary as it sounds
-          .build();
-
+          .setEnableMetrics(MetricType.DEBUG); // Not as scary as it sounds
+      if(config.enableTls()) {
+        // Build SSLContext
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init((KeyStore) null);
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, tmf.getTrustManagers(), null);
+        // Create the client in TLS mode
+        connectionFactoryBuilder.setSSLContext(sslContext);
+      }
+      if(Objects.equals(config.getClientMode(), "dynamic")) {
+        connectionFactoryBuilder.setClientMode(ClientMode.Dynamic);
+      }
+      else {
+        connectionFactoryBuilder.setClientMode(ClientMode.Static);
```

Comment:
we should throw an error if someone provides an invalid client mode rather than defaulting to static

Reply:
That change will not be backward compatible. As per my understanding, static mode mirrors the functionality present currently with spy.memcache. In case, the users do not provide any mode, we should default to static mode.

Reply:
I'm not saying we should throw an error when client mode is left out, we should still default to static if no client mode is provided. However, if an invalid client mode is provided we should throw an error.

Reply:
Updated